HTTP/Request2/Response.php

```diff
@@ -634,16 +634,31 @@ public static function decodeGzip($data)
 
         // don't pass $dataSize to gzinflate, see bugs #13135, #14370
         $unpacked = gzinflate(substr($data, $headerLength, -8));
+        if (false !== $unpacked) {
```

Comment:
The following stuff is precisely the reason I was reluctant to accept the original PR.

Come to think of it, can you test with something like (strlen($unpacked) & 0xffffffff) === ($dataSize & 0xffffffff) instead of using modulo?